apps/backoffice-v2/src/pages/Entity/hooks/useTasks/useTasks.tsx

```diff
@@ -232,6 +235,22 @@ export const useTasks = ({
           },
         ];
 
+  const postApproveEventName = getPostUpdateEventNameEvent(workflow);
+  const { mutate: mutateApproveTaskById, isLoading: isLoadingApproveTaskById } =
+    useApproveTaskByIdMutation(workflow?.id, postApproveEventName);
+  const onMutateApproveTaskById = useCallback(
+    ({
+        taskId,
+        contextUpdateMethod,
+      }: {
+        taskId: string;
+        contextUpdateMethod: 'base' | 'director';
+      }) =>
+      () =>
+        mutateApproveTaskById({ documentId: taskId, contextUpdateMethod }),
+    [mutateApproveTaskById],
+  );
```

Comment:
Hoisted logic from `useCallToActionLogic` - will be encapsulated at a later time.